resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java

```diff
@@ -47,12 +49,36 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * An Apache HTTP engine for use with the new Builder Config style.
  */
 public class ManualClosingApacheHttpClient43Engine implements ApacheHttpClientEngine
 {
+   private static class Closer implements Runnable {
+      private final AtomicBoolean closed;
+      private final CloseableHttpClient client;
+
+      private Closer(final AtomicBoolean closed, final CloseableHttpClient client) {
+         this.closed = closed;
+         this.client = client;
+      }
+
+      @Override
+      public void run() {
+         if (closed.compareAndSet(false, true)) {
+            if (client != null) {
+               LogMessages.LOGGER.closingForYou(this.getClass());
```

Comment:
Just to make sure it would not be forgotten, following [this discussion](https://github.com/eclipse/microprofile-rest-client/issues/348) we may need to avoid this log message if connection is for a CDI-created MicroProfile rest client (depending on the way it would be clarified by the spec in future versions).

Reply:
IMO that is a bug in the MP REST Client specification. The [`Client`](https://jakarta.ee/specifications/restful-ws/3.1/apidocs/jakarta.ws.rs/jakarta/ws/rs/client/client) API specifically says:

> Client instances must be [properly closed](https://jakarta.ee/specifications/restful-ws/3.1/apidocs/jakarta.ws.rs/jakarta/ws/rs/client/client#close()) before being disposed to avoid leaking resources.

I think the MP REST Client spec needs to clarify itself. There should be a way to do it, but I've not looked at the spec itself in a while.

Reply:
Actually, it looks like resteasy-microprofile handles creating the client, so I would likely say it's a bug there. I apologize for not looking close enough. I filed https://github.com/resteasy/resteasy-microprofile/issues/50